doc/ast-traversal.mld

```diff
@@ -0,0 +1,113 @@
+{%html: <div style="display: flex; justify-content:space-between"><div>%}{{!"matching-code"}< Destructing AST nodes}{%html: </div><div>%}{{!"good-practices"}Good practices >}{%html: </div></div>%}
+
+{0 AST traversals}
+
+The {!Ppxlib.Parsetree} is a very complex type. Other Ppxlib modules such as
+Metaquot, Ast_builder and Ast_pattern help in generating and matching values,
+but only when the overall structure of the code is known in advance.
+
+For other use cases, such as extracting all identifiers, checking that a
+property is verified, on replacing all integer constants by something else,
+those modules cannot really help. All these examples relates with another kind
+of Parsetree manipulations, known as traversals.
+
+A traversal is a recursive function which will be called on a value and all
+subvalues for a certain type. For instance, [List.map] is a traversal of the
+[list] type. In the case of a [list], a map is very simple to write, but in the
+case of the very complex [Parsetree] type, it is a lot of code! Fortunately,
+Ppxlib provides way to ease this.
```

Comment:
a style nit: "Ppxlib" renders very badly. Perhaps you should refer to it consistently as ppxlib rather than capitalising it?

Reply:
You are right! I checked how it was added [here](https://github.com/ocaml/ocaml.org/pull/535), so I replaced every occurrence of Ppxlib to `ppxlib`.